src/shared/featureController.ts

```diff
@@ -5,70 +5,84 @@
 
 import { SettingsConfiguration } from './settingsConfiguration'
 
-// This array holds keys for session-permanent settings.
-// These settings are set as permanent at extension load.
-// Changing these keys' values in settings.json during the extension's lifecycle
-//     will not reflect until the next time the extension loads.
-const sessionPermanentSettings: string[] = [
-]
+// This enum represents the list of active feature flags to check for
+// Features all need to be added with matching enums and string values
+// e.g. `Feature1 = 'Feature1'`
+// Features that do not follow this scheme will not work.
+// You cannot have more active features than FeatureController.maxFeatures (default: 5)
+// Any features that are flagged in the code but not added here will always return false.
+export enum ActiveFeatureKeys {
+}
 
 /**
- * This class handles feature access for unreleased features.
- * Access can be granted to a feature by adding a boolean flag to the VS Code settings.json file.
- * Features can be designated as session-permanent: this means that the setting won't change if altered mid-session
- * To designate a feature as session-permanent, add the feature to the sessionPermanentSettings array in this file.
+ * This class handles feature access for unreleased or gated features.
+ * Example: adding a feature called `NewFeature1`:
+ *
+ * Add the following to the `featureController.ActiveFeatureKeys` enum:
  *
- * Example: use the following setting to add a feature called `newFeature1`:
- * {
- * ...
- *     "aws.toggle.newFeature1": true,
- * ...
+ * ```
+ * export enum ActiveFeatureKeys {
+ *     NewFeature1 = 'NewFeature1'
  * }
- * This feature can then be accessed by calling FeatureController.isFeatureActive('newFeature1')
+ * ```
+ *
+ * You can then gate your feature with the following code (returning a boolean):
+ *
+ * ```
+ * FeatureController.isFeatureActive(featureController.ActiveFeatureKeys.NewFeature1)
+ * ```
+ *
+ * Finally, you can access the gated feature by adding the following snippet to the VS Code `settings.json` file:
+ *
+ * ```
+ *     "aws.experimentalFeatureFlags": [
+ *         "NewFeature1",
+ *     ],
+ * ```
  */
 export class FeatureController {
 
-    private readonly permanentSettingsMap: Map<string, boolean>
+    private readonly permanentSettings: Set<string>
```

Comment:
Consider naming this to something more intuitive like `enabledFeatures`. The current naming is caught up in the initial design.